### PR TITLE
[Core] Clean up includes on model_part.h

### PR DESCRIPTION
**Pull request by loumalouomega**
These includes are already added by the rest of dependencies

### CODE REVIEW

File: kratos/includes/model_part.h [hunk 1]
```diff
@@ -11,42 +11,20 @@
...
 
 // External includes
 
-
 // Project includes
-#include "includes/define.h"
 #include "includes/serializer.h"
 #include "includes/process_info.h"
-#include "containers/data_value_container.h"
-#include "includes/mesh.h"
```
Comment (philbucher) [inline kratos/includes/model_part.h]: where is mesh included such that it also works without?
Could this be another cotire problem?

**Comment by loumalouomega** [comment]
I compiled without cotire, so it should work

File: kratos/includes/model_part.h (hunk 1 above)
Comment (philbucher) [reply kratos/includes/model_part.h]: Just checked, `communicator.h` includes (and uses) it

**Comment by pooyan-dadvand** [comment]
Seems ok to me
The master slave is included in mesh?

**Comment by loumalouomega** [comment]
> Seems ok to me
> The master slave is included in mesh?

Indeed

File: kratos/includes/model_part.h [hunk 2]
```diff
@@ -11,42 +11,20 @@
...
-#include <sstream>
-#include <cstddef>
-
 
 // External includes
 
-
 // Project includes
-#include "includes/define.h"
 #include "includes/serializer.h"
```
Comment (philbucher) [inline kratos/includes/model_part.h:26]: can this be removed?

File: kratos/includes/model_part.h [hunk 3]
```diff
@@ -11,42 +11,20 @@
...
 #include "includes/process_info.h"
-#include "containers/data_value_container.h"
-#include "includes/mesh.h"
-#include "includes/element.h"
-#include "includes/condition.h"
 #include "includes/communicator.h"
 #include "includes/table.h"
-#include "containers/pointer_vector_map.h"
 #include "containers/pointer_hash_map_set.h"
 #include "input_output/logger.h"
```
Comment (philbucher) [inline kratos/includes/model_part.h:37]: what abt this?

Comment (loumalouomega) [reply kratos/includes/model_part.h:37]: It is already included in other files, don't remember which one

**Review by roigcarlo — CHANGES_REQUESTED** [review]
@loumalouomega I do not agree with the changes in general (even while particular includes that have been removed are ok).

As a rule of thumb, we should not remove any include that contains any class/definition etc... that is being used directly by the class or classes the header is defining. This does not apply if there is a forward declaration, but is not the case. TL;DR the header should compile as a stand-alone file without having to rely in includes from includes.

In particular `mesh.h` and `data_value_container.h` (and others) should not be removed as are being used directly by the modelpart ( and even deriving from one! )

Same with `define.h`.

Please re add the necessary ones before merging.

**Comment by loumalouomega** [comment]
@roigcarlo Ok you are right, I did basically what this static analysis software do, but I am not a machine...or am I?

**Comment by loumalouomega** [comment]
Ping @roigcarlo

**Comment by roigcarlo** [comment]
@loumalouomega There are still plenty of includes needed by the file that are deleted.

```C++
- #include <string>
- #include <iostream>
- #include <sstream>
- #include "includes/element.h"
- #include "includes/condition.h"
```
These are needed

```C++
- #include "containers/variable.h"
- #include "containers/variable_component.h"
- #include "containers/vector_component_adaptor.h"
- #include "containers/variable_data.h"
```
These you can remove them but I think we should replace them with `#include "containers/variable_list.h"`

```C++
- #include "includes/master_slave_constraint.h"
```
This one I am not sure, maybe it is possible to change the `MasterSlaveConstraint::Pointer` (line 715) for something depending on `MeshType`

**Comment by loumalouomega** [comment]
> @loumalouomega There are still plenty of includes needed by the file that are deleted.
> 
> ```c++
> - #include <string>
> - #include <iostream>
> - #include <sstream>
> - #include "includes/element.h"
> - #include "includes/condition.h"
> ```
> 
> These are needed
> 
> ```c++
> - #include "containers/variable.h"
> - #include "containers/variable_component.h"
> - #include "containers/vector_component_adaptor.h"
> - #include "containers/variable_data.h"
> ```
> 
> These you can remove them but I think we should replace them with `#include "containers/variable_list.h"`
> 
> ```c++
> - #include "includes/master_slave_constraint.h"
> ```
> 
> This one I am not sure, maybe it is possible to change the `MasterSlaveConstraint::Pointer` (line 715) for something depending on `MeshType`

Ok

**Comment by roigcarlo** [comment]
I and @loumalouomega agreed to close this for lack of consensus

**Pull request closed without merging**